src/sagas/atomicSwap.js

```diff
@@ -76,14 +87,91 @@ function* fetchProposalData(action) {
             return;
         }
 
-        // TODO: Implement the actual communication with the backend
-        throw new Error(`Proposal fetching not implemented.`)
+        // Fetch data from the backend
+        const responseData = yield swapService.get(proposalId, password);
+        yield put(proposalFetchSuccess(proposalId, responseData));
+
+        // On success, build the proposal object locally and enrich it
+        const wallet = yield select((state) => state.wallet);
+        const newData = generateReduxObjFromProposal(
+          proposalId,
+          password,
+          responseData.partialTx,
+          wallet,
+        );
+
+        // Adding the newly generated metadata to the proposal
+        const enrichedData = { ...responseData, ...newData.data };
+        yield put(proposalUpdated(proposalId, enrichedData));
+    } catch (e) {
+        let errorMessage;
+        const backendErrorData = e.response?.data || {};
+        switch (backendErrorData.code) {
+            case ATOMIC_SWAP_SERVICE_ERRORS.ProposalNotFound:
+                errorMessage = t`Proposal not found.`;
+                break;
+            case ATOMIC_SWAP_SERVICE_ERRORS.IncorrectPassword:
+                errorMessage = t`Incorrect password.`;
+                break;
+            default:
+                errorMessage = t`An error occurred while fetching this proposal.`;
+        }
+        yield put(proposalFetchFailed(proposalId, errorMessage));
+    }
+}
+
+/**
+ * Makes the request to the backend to create a proposal and returns its results via saga events
+ * @param {string} action.partialTx
+ * @param {string} action.password
+ */
+function* createProposalOnBackend(action) {
+    const { password, partialTx } = action;
+
+    try {
+        // Cleaning up the error handling redux object
+        yield put(lastFailedRequest(undefined))
+
+        // Request an identifier from the service backend
+        const { success, id: proposalId } = yield swapService.create(partialTx, password);
+
+        // Error handling
+        if (!success) {
+            yield put(lastFailedRequest({
+                message: t`An error occurred while creating this proposal.`
+            }))
+            return;
+        }
+
+        // Generate a minimal redux object to allow for the fetch request
+        yield(put(importProposal(proposalId, password)));
+
+        // Fetch the full proposal data from the backend
+        const fetchProposalResponse = yield call(
+          dispatchAndWait,
+          proposalFetchRequested(proposalId, password),
+          specificTypeAndPayload(
+            [ types.PROPOSAL_FETCH_SUCCESS ],
+            { proposalId: proposalId })
+          ,
+          specificTypeAndPayload(
+            [ types.PROPOSAL_FETCH_FAILED ],
+            { proposalId: proposalId })
+          ,
+        );
+
+        // Error handling
+        if (fetchProposalResponse.falure) {
```

Comment:
This is a typo on an existing code, will be fixed application-wide on the next PR